lfshttp/client.go

```diff
@@ -346,6 +346,24 @@ func (c *Client) doWithRedirects(cli *http.Client, req *http.Request, remote str
 	return c.doWithRedirects(cli, redirectedReq, remote, via)
 }
 
+func (c *Client) configureProtocols(u *url.URL, tr *http.Transport) error {
+	version, _ := c.uc.Get("http", u.String(), "version")
+	switch version {
+	case "HTTP/1.1":
+		// This disables HTTP?2, according to the documentation.
+		tr.TLSNextProto = make(map[string]func(authority string, c *tls.Conn) http.RoundTripper)
+	case "HTTP/2":
+		if u.Scheme != "https" {
+			return fmt.Errorf("HTTP/2 cannot be used except with TLS")
+		}
+		http2.ConfigureTransport(tr)
+		delete(tr.TLSNextProto, "http/1.1")
+	case "":
```

Comment:
I'm not too familiar with the `URLConfig.Get()` method (well, or any of the LFS client code! :-) but is it possible it might return some other non-empty value here, perhaps due to a misconfiguration?  I just wondered about a `default:` here instead ...

Reply:
That method returns either `value, true`, where `value` is the value that's from the config file (which can either be `http.version` or `http.MATCHING-URL.version`, as appropriate), or `"", false` if no such value exists.  In other words, this is the same return value as a map would produce.

I agree that a default is appropriate here, since we should probably return an error in that case; thanks for the suggestion.  Will fix.